fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java

```diff
@@ -3,75 +3,118 @@
-import com.starrocks.cluster.ClusterNamespace;
 import com.starrocks.common.AnalysisException;
 import com.starrocks.common.Config;
 import com.starrocks.common.ErrorCode;
 import com.starrocks.common.ErrorReport;
 import com.starrocks.common.FeConstants;
 import com.starrocks.common.FeNameFormat;
+import com.starrocks.common.util.PropertyAnalyzer;
 import com.starrocks.external.elasticsearch.EsUtil;
-import com.starrocks.mysql.privilege.PrivPredicate;
 import com.starrocks.qe.ConnectContext;
-import com.starrocks.server.GlobalStateMgr;
-import com.starrocks.sql.ast.Relation;
+import com.starrocks.sql.common.MetaUtils;
 import org.apache.commons.collections.CollectionUtils;
 
 import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.TreeSet;
 import java.util.stream.Collectors;
 
 import static com.starrocks.catalog.AggregateType.BITMAP_UNION;
+import static com.starrocks.catalog.AggregateType.HLL_UNION;
 
 public class CreateTableAnalyzer {
     private static final Set<String> engineNames;
+    private static Set<String> charsetNames;
+    private static final String DEFAULT_CHARSET_NAME = "utf8";
 
     static {
         engineNames = Sets.newHashSet();
         engineNames.add("olap");
         engineNames.add("mysql");
+        engineNames.add("broker");
         engineNames.add("elasticsearch");
         engineNames.add("hive");
+        engineNames.add("iceberg");
+        engineNames.add("hudi");
+        engineNames.add("jdbc");
     }
 
-    public static void transformCreateTableStmt(CreateTableStmt createTableStmt, ConnectContext session) {
-        createTableStmt.setClusterName(session.getClusterName());
-        TableName tableName = createTableStmt.getDbTbl();
-        analyzeTableName(tableName, session);
+    static {
+        charsetNames = Sets.newHashSet();
+        charsetNames.add("utf8");
+        charsetNames.add("gbk");
+    }
+
+    private static void analyzeEngineName(String engineName) {
+        if (Strings.isNullOrEmpty(engineName)) {
+            engineName = "olap";
+        }
+        engineName = engineName.toLowerCase();
+
+        if (!engineNames.contains(engineName)) {
+            throw new SemanticException("Unknown engine name: %s", engineName);
+        }
+    }
 
-        FeNameFormat.verifyTableName(tableName.getTbl());
+    private static void analyzeCharsetName(String charsetName) {
+        if (Strings.isNullOrEmpty(charsetName)) {
+            charsetName = "utf8";
+        }
+        charsetName = charsetName.toLowerCase();
 
-        if (!GlobalStateMgr.getCurrentState().getAuth()
-                .checkTblPriv(ConnectContext.get(), tableName.getDb(), tableName.getTbl(), PrivPredicate.CREATE)) {
-            ErrorReport.reportSemanticException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "CREATE");
+        if (!charsetNames.contains(charsetName)) {
+            throw new SemanticException("Unknown charset name: " + charsetName);
```

Comment:
the format do not conform to line 74. better to use one format. format type is preferred for me